Warnings report post process actions

nessid opened the pull request:
## Changelog Description
This pull request introduces a warnings report and post-process actions in the `ayon-core` project.
The primary change is the addition of the `PublishValidationException` class, designed to handle validation errors and warnings during the publishing process (it serves as a parent to the pre-existing `PublishValidationError` class and the new `PublishValidationWarning` class). 
Actions are now listed after the plugin process to incorporate dynamically added actions based on the process results.
Additionally, there are updates to color codes and UI elements to enhance visual clarity.

## Additional info
The `PublishValidationException` class is now the base for specific validation error and warning classes, offering detailed error messages and descriptions using Markdown syntax. This change enhances error handling and user communication in the publishing workflow. Actions listed after the plugin process ensure that any actions added dynamically based on the process results are captured and executed appropriately.

nessid committed:
feat: warnings report / get actions post plugin process

- report warnings instead of only errors
- get actions by plugin post plugin process to get actions collected during process also
- slight icons and colors modifications

nessid committed:
feat: warnings report / get actions post plugin process

- report warnings instead of only errors
- get actions by plugin post plugin process to get actions collected during process also
- slight icons and colors modifications

nessid committed:
chore: add log levels and warning exception

nessid committed:
chore: color adjustments

remove "info" color and change log warning color to match exception warning

nessid committed:
refactor: PublishValidationReportItem class

add parent class to PublishValidationError and PublishValidationWarning

nessid committed:
chore: adjust colors and icons

adjust colors and icons depending on the plugin proces result

nessid committed:
refactor: extact _determine_icon_and_color from paintEvent

nessid committed:
chore: remove print

nessid committed:
fix: warned plugins

Adapt way to indentify warned plugins to the new way to declare them

nessid committed:
fix: warned plugins

Adapt way to indentify warned plugins to the new way to declare them

nessid committed:
refactor: use original method and variables names

use "error" back instead of "report_item", as now both error and warnings are Exceptions. This allows to stay closer to the original code

nessid committed:
chore: _on_publish_stop

add docstring

nessid committed:
chore: original names

nessid committed:
feat: warned

add warned attribute to LogicItam and InstanceItem to have the right icon on instances in Details tab

nessid committed:
fix: correct if statement

nessid committed:
feat: warned

add ITEM_WARNED_ROLE

nessid committed:
fix: correct errored plugin detection

nessid committed:
doc: correct docstring

nessid committed:
refactor: use original names

use original names from main branch when possible

nessid committed:
revert: original names

Use original names when it is possible

nessid committed:
revert: original names

Use original names when it is possible

nessid committed:
chore: rename variables

nessid committed:
chore: rename class and attributes

replace "report item" with "exception" and "report_type" with "exception_type"

nessid committed:
revert: match with main branch

EnisNewborn commented on client/ayon_core/style/__init__.py:
```diff
@@ -117,7 +117,7 @@ def get_objected_colors(*keys):
 
 def _load_stylesheet():
-    """Load strylesheet and trigger all related callbacks.
+    """Load stylesheet and trigger all related callbacks.
```
Not sure if it forth fork just for a description typo

nessid replied:
typo restored

EnisNewborn commented on client/ayon_core/tools/publisher/control.py:
```diff
@@ -423,7 +437,7 @@ class PublishPluginActionItem:
         plugin_id (str): Plugin id.
         active (bool): Action is active.
-        on_filter (str): Actions have 'on' attribte which define when can be
+        on_filter (str): Actions have 'on' attribute which define when can be
```
Same. Not sure it worth fix source code

nessid replied:
typo restored

EnisNewborn commented on client/ayon_core/tools/publisher/control.py:
```diff
@@ -518,11 +534,12 @@ def to_data(self):
     @classmethod
     def from_result(cls, plugin_id, error, instance):
-        """Create new object based on resukt from controller.
+        """Create new object based on result from controller.
```
Source code typo ;)

nessid replied:
typo restored

EnisNewborn commented on client/ayon_core/tools/publisher/widgets/images/thumb-up.png:
Please remove if not used 😁

nessid replied:
image removed

EnisNewborn commented on client/ayon_core/tools/publisher/widgets/report_page.py:
```diff
@@ -1803,7 +1988,7 @@ def _on_error_selection(self):
 
 class ReportPageWidget(QtWidgets.QFrame):
-    """Widgets showing report for artis.
+    """Widgets showing report for artists.
```
source typo ;)

nessid replied:
typo restored

EnisNewborn reviewed (commented):
First short overview. 
I didn't realized the amount of work here. Great job !

nessid committed:
fix: add error type indicators to PublishValidationErrorsReport

- Added `has_error_type_errors` and `has_warning_type_errors` parameters to the constructor of `PublishValidationErrorsReport`.
- Updated the docstring to include the new parameters.
- Modified the `from_data` method to deserialize `has_error_type_errors` and `has_warning_type_errors` from the provided data.
- Ensured the `to_data` method includes `has_error_type_errors` and `has_warning_type_errors` in the serialized output.

nessid committed:
revert: restore typos to align with the original codebase

The original codebase's typos were corrected to improve readability. To maintain alignment with the source repository, these changes are being reverted.

nessid committed:
chore: remove unused icon

Removed an icon that was added to the codebase but is no longer in use.

nessid committed:
revert: restore typos to align with the original codebase

The original codebase's typos were corrected to improve readability. To maintain alignment with the source repository, these changes are being reverted.

nessid committed:
revert: restore typos to align with the original codebase

The original codebase's typos were corrected to improve readability. To maintain alignment with the source repository, these changes are being reverted.

nessid committed:
revert: restore typos to align with the original codebase

The original codebase's typos were corrected to improve readability. To maintain alignment with the source repository, these changes are being reverted.

nessid committed:
feat: add custom colors to log icons and validation error icons

- Added log level colors for debug, info, warning, error, and critical log levels in the stylesheet.
- Implemented color change for validation error and warning icons in the report viewer.
- Updated `LogIconFrame` and `ValidateErrorIconFrame` to use added log-level colors.
- Improved the visibility management of log icons and validation error icons.
- Updated `settings.json` to include new color definitions for log levels and validation states.

nessid committed:
chore: remove unused import

nessid commented:
This pull request is being closed because a new branch, `plugins_view_actions`, has been created based on it. The new branch includes the features from the `warnings-report_post-process-actions` branch as well as new functionality.

The `plugins_view_actions` branch introduces actions available in the plugins list in the "Details" tab of the publisher. This feature allows actions not only for plugins with errors or warnings but for all plugins with actions.

The `warnings-report_post-process-actions` branch featured the handling of non-blocking publish errors (warnings) in the report, with the ability to ignore them and proceed with publishing. It also included adding actions to `PublishPluginsProxy` after the plugins have been processed, to account for actions created dynamically during the process based on the results.

A new pull request will be created for the `plugins_view_actions` branch, with all changes from this pull request, along with the new features.
